server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java

```diff
@@ -147,6 +147,15 @@ public int size() {
         return size;
     }
 
+    @Override
+    public BucketCardinality bucketCardinality() {
+        /*
+         * This might be a bit of a lie though because you can't really use
+         * pipeline aggregations on composite aggregations.
+         */
+        return BucketCardinality.MANY;
```

Comment:
Yeah... I wonder if we should set this to `NONE` for the time being, to prevent people from accidentally using pipelines on composite?  There's an issue flying around somewhere discussing it... but last time we chatted, we decided to punt on the decision.  Some pipelines are theoretically safe (bucket_script, operating on the bucket itself), but others like `derivative` are totally not kosher.

Reply:
Yeah. Composite is a weird beast.